sim/hunter/aspects.go

```diff
@@ -8,86 +8,188 @@ import (
+            if isMelee {
+                aura.Unit.MultiplyMeleeSpeed(sim, 1/bonusMultiplier)
+            } else {
+                aura.Unit.MultiplyRangedSpeed(sim, 1/bonusMultiplier)
+            }
+        },
+    })
+}
 
-	aspectOfTheHawkAura.NewExclusiveEffect("Aspect", true, core.ExclusiveEffect{})
+// Function to get the maximum attack power for Aspect of the Hawk based on character level
+func (hunter *Hunter) getMaxAspectOfTheHawkAttackPower(level int) float64 {
+    attackPower := [8]float64{0, 20, 35, 50, 70, 90, 110, 120} // Static data for attack power per rank
+    levels := [8]int{0, 10, 18, 28, 38, 48, 58, 60} // Levels at which ranks are available
 
-	return core.SpellConfig{
-		ActionID:      actionID,
-		Flags:         core.SpellFlagAPL,
-		Rank:          rank,
-		RequiredLevel: level,
+    maxAttackPower := 0.0
 
-		Cast: core.CastConfig{
-			DefaultCast: core.Cast{
-				GCD: core.GCDDefault,
-			},
-		},
-		ExtraCastCondition: func(sim *core.Simulation, target *core.Unit) bool {
-			return !aspectOfTheHawkAura.IsActive()
-		},
+    for rank := 1; rank <= 7; rank++ {
+        if level >= levels[rank] {
+            maxAttackPower = attackPower[rank]
+        }
+    }
+    return maxAttackPower
+}
 
-		ApplyEffects: func(sim *core.Simulation, _ *core.Unit, _ *core.Spell) {
-			aspectOfTheHawkAura.Activate(sim)
-		},
-	}
+// Configuration for Aspect of the Hawk spell
+func (hunter *Hunter) getAspectOfTheHawkSpellConfig(rank int) core.SpellConfig {
+    var impHawkAura *core.Aura
+    improvedHawkProcChance := 0.01 * float64(hunter.Talents.ImprovedAspectOfTheHawk)
+
+    spellIds := [8]int32{0, 13165, 14318, 14319, 14320, 14321, 14322, 25296}
+    levels := [8]int{0, 10, 18, 28, 38, 48, 58, 60}
+
+    spellId := spellIds[rank]
+    level := levels[rank]
+
+    if hunter.Talents.ImprovedAspectOfTheHawk > 0 {
+        impHawkAura = hunter.createImprovedHawkAura(
+            "Quick Shots",
+            core.ActionID{SpellID: 6150},
+            false, // Ranged
+        )
+    }
+
+    // Use utility function to get the attack power based on rank
+    rap := hunter.getMaxAspectOfTheHawkAttackPower(level)
+
+    actionID := core.ActionID{SpellID: spellId}
+    aspectOfTheHawkAura := hunter.NewTemporaryStatsAuraWrapped(
+        "Aspect of the Hawk"+strconv.Itoa(rank),
+        actionID,
+        stats.Stats{stats.RangedAttackPower: rap},
+        core.NeverExpires,
+        func(aura *core.Aura) {
+            aura.OnSpellHitDealt = func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+                if spell != hunter.AutoAttacks.RangedAuto() {
+                    return
+                }
+
+                if impHawkAura != nil && sim.RandomFloat("Imp Aspect of the Hawk") < improvedHawkProcChance {
+                    impHawkAura.Activate(sim)
+                }
+            }
+        })
+
+    aspectOfTheHawkAura.NewExclusiveEffect("Aspect", true, core.ExclusiveEffect{})
```

Comment:
Don't forget to add this exclusive effect to the aspect of the falcon code as well